llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -25434,6 +25436,41 @@ performScalarToVectorCombine(SDNode *N, TargetLowering::DAGCombinerInfo &DCI,
   return NVCAST;
 }
 
+static SDValue performHistogramCombine(SDNode *N,
```

Comment:
Rather than doing this as a target-specific DAG combine, this should be done as generic type legalisation in LegalizeVectorTypes.cpp. See `DAGTypeLegalizer::SplitVectorOperand` for how this is done for other operations.

Reply:
You're right. I've moved the splitting to `DAGTypeLegalizer::SplitVecOp_VECTOR_HISTOGRAM` and added/moved the `INC` operand promotion to `DAGTypeLegalizer::PromoteIntOp_VECTOR_HISTOGRAM`.